src/plugins/forecast.rs

```diff
@@ -8,26 +8,71 @@ pub struct ForecastPlugin {
 }
 
 impl ForecastPlugin {
-    pub fn new(darksky_api_key: String, maps_api_key: String) -> Arc<Self> {
-        Arc::new(ForecastPlugin {
+    pub fn new(darksky_api_key: String, maps_api_key: String) -> Self {
+        ForecastPlugin {
             darksky: darksky::Client::new(darksky_api_key),
             maps: maps::Client::new(maps_api_key),
-        })
+        }
     }
 }
 
 impl ForecastPlugin {
-    async fn lookup_weather(
-        self: Arc<Self>,
-        ctx: Arc<Context>,
-        location: ForecastLocation,
-    ) -> Result<()> {
+    async fn handle_weather(&self, ctx: &Context, arg: Option<&str>) -> Result<()> {
+        match self.extract_location(ctx, arg).await? {
+            LocationStatus::SingleLocation(location) => {
+                self.lookup_weather(ctx, location).await?;
+            }
+            LocationStatus::MultipleLocations(locations) => {
+                ctx.mention_reply(&format!(
+                    "Multiple possible locations. {}.",
+                    locations.into_iter().map(|loc| loc.address).join(", "),
```

Comment:
I think we should limit the number we're printing here.

Reply:
Good call